kratos/input_output/vtk_output.cpp

```diff
@@ -0,0 +1,445 @@
+                    rFileStream <<"\n";
+                }
+                for (const auto& condition : local_mesh.Conditions())
+                {
+                    Variable<double> element_result_variable = KratosComponents<Variable<double>>::Get(element_result_name);
+                    const auto &condition_result = condition.GetValue(element_result_variable);
+                    WriteScalarDataToFile(condition_result, rFileStream);
+                    rFileStream <<"\n";
+                }
+            }
+            else if (data_characteristic == VtkOutput::WriteDataType::VTK_VECTOR)
+            {
+                for (const auto& elem : local_mesh.Elements())
+                {
+                    Variable<array_1d<double, 3>> element_result_variable = KratosComponents<Variable<array_1d<double, 3>>>::Get(element_result_name);
+                    const auto &element_result = elem.GetValue(element_result_variable);
+                    WriteVectorDataToFile(element_result, rFileStream);
+                    rFileStream << "\n";
+                }
+
+                for (const auto& condition : local_mesh.Conditions())
+                {
+                    Variable<array_1d<double, 3>> element_result_variable = KratosComponents<Variable<array_1d<double, 3>>>::Get(element_result_name);
+                    const auto &condition_result = condition.GetValue(element_result_variable);
+                    WriteVectorDataToFile(condition_result, rFileStream);
+                    rFileStream << "\n";
+                }
+            }
+        }
+    }
+}
+
+void VtkOutput::WriteModelPart(const ModelPart &rModelPart)
+{
+    Initialize(rModelPart);
+
+    // Make the file stream object
+    std::string output_file_name = GetOutputFileName(rModelPart);
+    std::ofstream output_file;
+    if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII)
+    {
+        output_file.open(output_file_name, std::ios::out | std::ios::trunc);
+        output_file << std::scientific;
+        output_file << std::setprecision(mDefaultPrecision);
+    } else if (mFileFormat == VtkOutput::FileFormat::VTK_BINARY)
+    {
+        output_file.open(output_file_name, std::ios::out | std::ios::binary | std::ios::trunc);
+    }
+
+    WriteHeader(rModelPart, output_file);
+    WriteMesh(rModelPart, output_file);
+    WriteNodalResultsAsPointData(rModelPart, output_file);
+    WriteElementData(rModelPart, output_file);
+
+    output_file.close();
+}
+
+void VtkOutput::PrintOutput()
+{
+    //For whole model part
+    WriteModelPart(mrModelPart);
+    //For sub model parts
+    bool print_sub_model_parts = this->mOutputSettings["output_sub_model_parts"].GetBool();
+
+    if(print_sub_model_parts)
+    {
+        for (const auto& sub_model_part : mrModelPart.SubModelParts())
+        {
+            std::string output_file_name = GetOutputFileName(sub_model_part);
+            WriteModelPart(sub_model_part);
```

Comment:
if this would also call `PrintOutput` (which would have to accept a ModelPart => private overload) then this would automatically output all submodelparts and so on recursively

We can improve this in the future